docs/guide/mocking-requests.md

```diff
@@ -0,0 +1,60 @@
+# Mocking Requests
+
+When it comes to mocking network requests, our recommendation is to use [Mock Service Worker](https://mswjs.io/). It handles various types of network requests (like `REST` or `GraphQL`) and is completely framework agnostic.
+
+## Why
+
+We write tests in order to be able to ship with confidence. The more underlying behavior we mock, the less confidence we get since the system is less closer to real usage.
+
+Mock Service Worker (MSW) on the other hand, does not mock any underlying behavior. What it does instead, it uses [Service Worker API](https://developers.google.com/web/fundamentals/primers/service-workers) to intercept these requests.
+
+To read more about why to choose Mock Service Worker: [Introduction to MSW](https://mswjs.io/docs/).
+
+## Configuration
+
+Create a `handlers.ts` file to specify the requests that need intercepting
+
+```js
+import { rest } from 'msw'
+import { posts } from './src/mocks/posts'
+
+export const handlers = [
+  rest.get('url-to-be-captured', (req, res, ctx) => {
+    return res(ctx.status(200), ctx.json(posts))
+  }),
+]
+```
+
+Create a `server.ts` file to add the handlers to a MSW server
+
+```js
+import { setupServer } from 'msw/node'
+import { handlers } from './handlers'
+
+// add handlers to the MSW server
+export const server = setupServer(...handlers)
+```
+
+Add the following to your (test) setup file
+
+```js
+import { server } from './mocks/server'
+
+// Start server before all tests
+beforeAll(() => server.listen({ onUnhandledRequest: 'error' }))
+
+//  Close server after all tests
+afterAll(() => server.close())
+
+// Reset handlers after each test `important for test isolation`
+afterEach(() => server.resetHandlers())
+```
+You can find more information on the the `onUnhandledRequest` option [here](https://mswjs.io/docs/recipes/debugging-uncaught-requests)
```

Comment:
Per #279, the docs should be self-contained. If this is important enough to call out, we should describe it here (possibly as an aside) and link to the MSW docs a way for readers to see the full documentation.

Reply:
Removed link and explained it's purpose in a call-out. Do you feel this makes it better? Maybe the link itself could be added for them as a further reference.